src/README.md

```diff
@@ -398,6 +398,93 @@ print(f'Median from token to token duration: {np.median(durations):.2f} ms')
+
+It can be initialized from the path, in-memory IR representation or obtained from the `ov::genai::LLMPipeline` object.
+
+```cpp
+// Initialize from the path
+#include "openvino/genai/llm_pipeline.hpp"
+auto tokenizer = ov::genai::Tokenizer(models_path);
+
+// Get instance of Tokenizer from LLMPipeline.
+auto pipe = ov::genai::LLMPipeline pipe(models_path, "CPU");
+auto tokenzier = pipe.get_tokenizer();
+````
+
+```python
+import openvino_genai as ov_genai
+tokenizer = ov_genai.Tokenizer(models_path)
+
+// Or from LLMPipeline.
+pipe = ov_genai.LLMPipeline(models_path, "CPU")
+tokenizer = pipe.get_tokenizer()
+```
+
+`Tokenizer` has `encode` and `decode` methods which support the following arguments: `add_special_tokens`, `skip_special_tokens`, `padding_mode`, `max_length` arguments.
+
+In order to disable adding special tokens do the followings, in C++:
+```cpp
+auto tokens = tokenizer.encode("The Sun is yellow because", ov::genai::add_special_tokens(false));
+```
+
+In Python:
+```python
+tokens = tokenizer.encode("The Sun is yellow because", add_special_tokens=False)
+```
+
+Example of getting padded outputs:
+```cpp
+#include "openvino/genai/llm_pipeline.hpp"
+auto tokenizer = ov::genai::Tokenizer(models_path);
+std::vector<std::string> prompts = {"The Sun is yellow because", "The"};
+
+// With LONGEST string will be padded to the lengh of the longest sequence, max_length=1024 will be ignored.
+tokens = tokenizer.encode(prompts, ov::genai::padding_mode(ov::genai::PaddingMode::LONGEST), ov::genai::max_length(1024))
+// out_shape: [2, 6]
+
+// With MAX_LENGTH string are short and resulting tokens sequence will be padded to 1024.
+tokens = tokenizer.encode(prompts, ov::genai::padding_mode(ov::genai::PaddingMode::MAX_LENGTH), ov::genai::max_length(1024))
+// out_shape: [2, 1024]
+
+// With TRUNCATE long string will be truncated to the same 1024 length for the longer sequences 
+// and padded to the same 1024 length for the shorter sequences.
+tokens = tokenizer.encode(
+    ["This is THE LONGESTS string ever of all times " * 1000, "The"], 
+    ov::genai::padding_mode(ov::genai::PaddingMode::TRUNCATE), 
+    ov::genai::max_length(1024)
+)
+// out_shape: [2, 1024]
+```
+
+```python
+import openvino_genai as ov_genai
+
+tokenizer = ov_genai.Tokenizer(models_path)
+prompts = ["The Sun is yellow because", "The"]
+
+# String will be padded to the lengh of the longest sequence, max_length=1024 will be ignored.
+tokens = tokenizer.encode(prompts, padding_mode=ov_genai.PaddingMode.LONGEST, max_length=1024)
+print(tokens.input_ids.shape)
+# out_shape: [2, 6]
+
+# String are short and resulting tokens sequence will be padded to 1024.
+tokens = tokenizer.encode(prompts, padding_mode=ov_genai.PaddingMode.MAX_LENGTH, max_length=1024)
+print(tokens.input_ids.shape)
+# out_shape: [2, 1024]
+
+# Very long string will be truncated to the same 1024 length for the longer sequences 
+# and padded to the same 1024 length for the shorter sequences.
```

Comment:
> We should add that truncation is the default behaviour and the max_length value comes from the original tokenizer if not explicitly defined.

Same for docstrings

Reply:
Simplified regimes. Now we have 

- `max_length` which by default defines truncation value and padding to the longest sequence in the batch is also performed. (If `max_length` is not set, default value is taken from IR).

- and `pad_to_max_length` flag, which enables padding to the `max_length` instead of the longest sequence in the batch. 

If resulting sequence exceeds `max_length` it will be still truncated. So now we always have truncation, if someone need to completely disable it they can set it to int32_max.

@apaniukov please take a look once more to the latest version, feed free to withdraw your approve if you don't agree with the latest version :)